packages/form-core/src/choice-group/ChoiceInputMixin.js

```diff
@@ -189,14 +189,28 @@ const ChoiceInputMixinImplementation = superclass =>
 
     connectedCallback() {
       super.connectedCallback();
+      this._labelNode.addEventListener('click', this._preventDuplicateLabelClick.bind(this));
       this.addEventListener('user-input-changed', this.__toggleChecked);
     }
 
     disconnectedCallback() {
       super.disconnectedCallback();
+      this._labelNode.removeEventListener('click', this._preventDuplicateLabelClick.bind(this));
       this.removeEventListener('user-input-changed', this.__toggleChecked);
     }
 
+    /**
+     * @param {Event} ev
+     */
+    // eslint-disable-next-line no-unused-vars
+    _preventDuplicateLabelClick(ev) {
```

Comment:
Looks good. Maybe add some explanation which 'platform hack' we work around here